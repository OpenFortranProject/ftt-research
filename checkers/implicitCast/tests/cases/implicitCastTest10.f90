! Test that we detected implicit cast
! when assigning function return value
! in the presence of explicit interface

function f()
  integer :: f
  real :: x = 42.4

  f = x
end function

program test
  interface
    function f()
    integer :: f
    end function
  end interface

  INTEGER :: x
  
  x = f() + 4
  
  print *, x

end
