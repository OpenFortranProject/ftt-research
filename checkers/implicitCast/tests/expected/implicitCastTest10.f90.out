Warning: environment variable COMPASS_PARAMETERS is not set, trying to use compass_parameters of the current path
 Running Prerequisite SgProject
ImplicitCast: /home/dagit/research/checkers/implicitCast/tests/cases/implicitCastTest10.f90:9.2: Implicit cast in Fortran code found:  expr 'x' from real to integer
